### ISSUE TITLE
default_input_processor with quoted markup in attribute

### ISSUE THREAD

**Issue by andrewbaxter** [scrapinghub/scrapylib#42]
It looks like default_input_processor generates strange results if &amp;gt; and the like appear in an attribute.

Rearranging the composition as this solves the problem:
```Python
default_input_processor = MapCompose(
    replace_br, remove_tags, unquote_markup, replace_escape, strip, clean_spaces
)
```

Example html:
```HTML
<span id="discount_box" data-option-text="Discount &lt;span class=&quot;percent&quot; id=&quot;buy_percent&quot;&gt;&lt;/span&gt;&lt;span class=&quot;percent&quot;&gt;%&lt;/span&gt;" data-default-text="up to &lt;span class=&quot;percent&quot; id=&quot;buy_percent&quot;&gt;54&lt;/span&gt;&lt;span class=&quot;percent&quot;&gt;%&lt;/span&gt;"  class="discount">up to <span class="percent" id="buy_percent">54</span><span class="percent">%</span></span>
```
Current result:
```Python
[u'%" data-default-text="up to 54%" class="discount">up to 54%']
```

**Comment by horva** [comment]
Hi, there is another case of unexpected content processing.
```python
html = """<p>&lt;&lt; ...The Sunnywale, Calif.-based internet giant announced plans to eliminate 2,000 jobs, or about 14% of its workforce of 14,000.<br><br>&#8220;Today&#8217;s actions are an important next step toward a bold, new Yahoo! &#8212; smaller, nimbler, more profitable and better equipped to innovate as fast as our customers and our industry require,&#8221; CEO Scott Thompson, who joined Yahoo in January, said in a statement. ... &gt;&gt;</p>"""
```
And when processed with default_input_processor, it returns
```python
default_input_processor(html)
[u'>']
```
But if we define default_input_processor with
```python
default_input_processor = MapCompose(remove_tags, unquote_markup, replace_br, replace_escape, strip, clean_spaces)
```
then result is
```python
[u'<< ...The Sunnywale, Calif.-based internet giant announced plans to eliminate 2,000 jobs, or about 14% of its workforce of 14,000.\u201cToday\u2019s actions are an important next step toward a bold, new Yahoo! \u2014 smaller, nimbler, more profitable and better equipped to innovate as fast as our customers and our industry require,\u201d CEO Scott Thompson, who joined Yahoo in January, said in a statement. ... >>']
```

**Comment by dangra** [comment]
fixed by #54
